### ISSUE TITLE
Decorating a bird phylogeny with a numerical variable

### ISSUE THREAD

**Issue by Sicily-F** [ArtPoon/ggfree#27]
![image](https://user-images.githubusercontent.com/68450039/112673419-9b62a200-8e5c-11eb-8a7e-175204760f72.png)

Hi there,
Thank you for this, your code is amazing - I've been trying to use ggtree and have found the documentation really poor, for my particular task. 

I'm trying to decorate a bird phylogeny (made with birdtree.org), and I'd like to decorate with another variable. My variable is the accuracy per species in an identification test, which I'd like to show with the phylogeny.

print(acc)

 [1] 0.9277778 0.9438059 0.9272353 0.8306878 0.8876118 0.9082126 0.9476373 0.9053498 0.9333333
[10] 0.9449735 0.9487179 0.9656085 0.9317739 0.9272353 0.9444444 0.9225589 0.9382716 0.9469136
[19] 0.9583333

As you can see, it's not an integer. This was the code I used 
`require(devtools)
devtools::install_github("ArtPoon/ggfree")

library(ggfree)
require(RColorBrewer)
L <- tree.layout(bird_tree, type='o')
plot(L, cex.lab=0.7, offset=2, mar=rep(5,4), col='chocolate')
image(L, z=as.matrix(acc), xlim=c(28.5,30), col=pal)`

However, the variable is misplaced inside the actual phylogeny

**Comment by ArtPoon** [comment]
Hi @Sicily-F, sorry for the late reply - I missed this issue notification somehow.  I will try to reproduce this issue.  Are you still encountering this problem?

**Comment by ArtPoon** [comment]
Ok I think I reproduced this problem and have some test code for you to try:
```R
require(ggfree)

# generate a random coalescent tree with 19 tips
set.seed(4)
phy <- rcoal(19)
L <- tree.layout(phy, 'o')

# user numeric vector from #27
acc <- c(0.9277778, 0.9438059, 0.9272353, 0.8306878, 0.8876118, 0.9082126, 
         0.9476373, 0.9053498, 0.9333333, 0.9449735, 0.9487179, 0.9656085, 
         0.9317739, 0.9272353, 0.9444444, 0.9225589, 0.9382716, 0.9469136,
         0.9583333)

# trying to reproduce the issue
pal <- hcl.colors(10, 'Blues')  # no palette specified in issue
plot(L, cex.lab=0.7, offset=0, mar=rep(3,4), col='chocolate')
axis(side=1, col=rgb(0,0,0,0.2), col.axis=rgb(0,0,0,0.2), cex.axis=0.8)
image(L, z=as.matrix(acc), xlim=c(0.66, 0.7), col=pal)
```
![image](https://user-images.githubusercontent.com/1109328/115100291-a9f72300-9f09-11eb-8dcd-f44bfb0950b8.png)

The main problem here was figuring out the scale of the circular tree to draw your "decorations".  To illustrate, I have drawn an axis in grey on top of the example tree above.  Note that it spans about ±0.6 units (branch length) away from the centre.  I used trial and error to determine where to draw the blocks and settled on start and stop radii of `c(0.66, 0.7)`.  Sorry that's not very intuitive or clear.  I'll fix the documentation.

**Comment by ArtPoon** [comment]
Since it's always a good idea to double-check how data are mapped to the tips:
```R
> cbind(phy$tip.label, acc)
            acc        
 [1,] "t2"  "0.9277778"
 [2,] "t3"  "0.9438059"
 [3,] "t17" "0.9272353"
 [4,] "t7"  "0.8306878"
 [5,] "t14" "0.8876118"
 [6,] "t16" "0.9082126"
 [7,] "t5"  "0.9476373"
 [8,] "t8"  "0.9053498"
 [9,] "t13" "0.9333333"
[10,] "t12" "0.9449735"
[11,] "t4"  "0.9487179"
[12,] "t19" "0.9656085"
[13,] "t1"  "0.9317739"
[14,] "t15" "0.9272353"
[15,] "t9"  "0.9444444"
[16,] "t18" "0.9225589"
[17,] "t6"  "0.9382716"
[18,] "t10" "0.9469136"
[19,] "t11" "0.9583333"
```
So the lowest accuracy values are being mapped to the darkest shades of blue, which is a bit counter-intuitive to me.  While I was looking this over, I realized that those white cells in the `image` output don't make sense, so I added a `cut` function to map the numeric vector to the integer range of the palette:
```R
pal <- hcl.colors(10)
col.map <- as.integer(cut(acc, 10))
plot(L, cex.lab=0.7, offset=0, mar=rep(3,4), col='chocolate')
axis(side=1, col=rgb(0,0,0,0.2), col.axis=rgb(0,0,0,0.2), cex.axis=0.8)
image(L, z=as.matrix(col.map), xlim=c(0.66, 0.7), col=pal)
```

**Comment by ArtPoon** [comment]
Great, I hope it works out for you - please feel free to re-open the issue if you encounter any problems.

**Comment by Sicily-F** [comment]
Hi there, 
I have tried the code, though the colour palette didn't seem to work; for the last two lines with the axis and image arguments, though I got the following warning: 

In doTryCatch(return(expr), name, parentenv, handler) :
  "offset" is not a graphical parameter

Thank you so much for all your help - I really appreciate it! 

![image](https://user-images.githubusercontent.com/68450039/115521075-b0c3c400-a282-11eb-9feb-e802191c6396.png)

**Comment by ArtPoon** [comment]
Hm, that warning message should go away if you update and re-install ggfree.
Looks like the horizontal scale of your tree ranges from about -80 to +80.  I also updated the `image` function to guess reasonable default values, so can you try updating the package first please?

**Comment by Sicily-F** [comment]
> Hm, that warning message should go away if you update and re-install ggfree.
> Looks like the horizontal scale of your tree ranges from about -80 to +80. I also updated the `image` function to guess reasonable default values, so can you try updating the package first please?

hi - absolutely, I'll try that

**Comment by Sicily-F** [comment]
> Hm, that warning message should go away if you update and re-install ggfree.
> Looks like the horizontal scale of your tree ranges from about -80 to +80. I also updated the `image` function to guess reasonable default values, so can you try updating the package first please?

hi - absolutely, I'll try that

**Comment by Sicily-F** [comment]
So I restarted, removed the package and redownloaded from the Githubrepo, but it now has no warning but still no colours on the plot - I'll keep trying

![image](https://user-images.githubusercontent.com/68450039/115883634-ebc22500-a445-11eb-8c15-e7fd90961203.png)

**Comment by ArtPoon** [comment]
I think the problem is that you have set `xlim` to be very small relative to the scale of your tree.  You're asking R to draw blocks with a width of about 0.04 units (0.7 minus 0.66), but the scale of your tree is over 100 units (see the gray horizontal axis).  So, the first thing I would try is to remove the `xlim` argument altogether - that will cause `image` to use default settings.  Then I suggest experimenting with different values of `xlim` such as `c(60, 80)` to start.

**Comment by Sicily-F** [comment]
Hi - thank you SO much - I'll need to edit it a bit in Inkscape, but it definitely shows what I want it to now, thank you again, I'll be sure to reference and share the repo

**Comment by Sicily-F** [comment]
hi again - I'm still having some slight issues interpreting the legend in terms of the range from -50 to 50, as I am not sure what a negative branch length would be?

**Comment by ArtPoon** [comment]
Hi @Sicily-F - sorry for the late reply, I did see your posting but when I don't respond to things right away they get buried under other e-mails!

When I am drawing a radial/circular tree, I set up the plot region so that the tree is centered on the origin (x=0, y=0).  The range of the plot region (minimum and maximum values for x and y) are determined by the branch lengths of the tree itself.  Now, because we are drawing branches radiating in all directions away from the center, some of them take negative values of x and/or y.  This does not mean that the branch has a negative length!

I hope that clarifies things.
